internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go

```diff
@@ -377,12 +377,12 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 }
 
 func resourceDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// There is no resource to delete a backup compliance policy, it can only be updated.
-	log.Printf("[WARN] Note: Deleting a Backup Compliance Policy resource in Terraform does not remove the policy from your Atlas Project. " +
-		"To disable a Backup Compliance Policy, the security or legal representative specified for the Backup Compliance Policy must contact " +
-		"MongoDB Support and complete an extensive verification process. ")
-
-	d.SetId("")
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
```

Comment:
Question - did we check if the error is clear from the API endpoint if we run delete and it's not marked as deletable?  Not quite our purview but we can push back on the creating team if it's not clear (since that could be really confusing for Terraform users if it's not).

Reply:
Good point, verified that the API does provide a meaningful error message:

> Error: error disabling the Backup Compliance Policy: **343f: https://cloud.mongodb.com/api/atlas/v2/groups/**343f/backupCompliancePolicy DELETE: HTTP 400 Bad Request (Error code: "CANNOT_DISABLE_BACKUP_COMPLIANCE_POLICY") Detail: Cannot disable Backup Compliance Policy because Disabling BCP requires MongoDB support or removing all clusters & retained snapshots. This can only be edited by filing a support ticket.

Reply:
That's great, thank you for checking!